gevent/_socket2.py

```diff
@@ -290,7 +290,7 @@ def send(self, data, flags=0, timeout=timeout_default):
         try:
             return sock.send(data, flags)
         except error as ex:
-            if ex.args[0] != EWOULDBLOCK or timeout == 0.0:
+            if ex.args[0] != EWOULDBLOCK or self.timeout == 0.0:
```

Comment:
I don't believe this change is correct. `timeout` is a keyword argument to this function; if it's isn't set, it's already replaced with `self.timeout`:
```
    def send(self, data, flags=0, timeout=timeout_default):
        sock = self._sock
        if timeout is timeout_default:
            timeout = self.timeout
        try:
           ...
```